api/v1/controllers/tokens.js

```diff
@@ -87,27 +86,22 @@ module.exports = {
    */
   postToken(req, res, next) {
     const resultObj = { reqStartTime: new Date() };
+    let tokenValue;
 
-    // req.user is set when verifying token with user details. If req.user is
-    // not set, then return error.
-    if (!req.user || !req.user.name) {
-      const tokenErr = new apiErrors.LoginError({
-        explanation: 'Unable to parse token. Please make sure that you ' +
-         'include an Authorization header (i.e. Authorization=YOUR_TOKEN) ' +
-         'with your POST request.',
-      });
-      tokenErr.resource = resourceName;
-      return u.handleError(next, tokenErr, resourceName);
-    }
-
-    // create token to be returned in response.
-    const tokenName = req.swagger.params.queryBody.value.name;
-    const tokenValue = jwtUtil.createToken(tokenName, req.user.name);
+    // get user details from req
+    authUtils.getUser(req)
+    .then((user) => {
+      // create token to be returned in response.
```

Comment:
Sorry, I am confused. Do you mean the "create" in comment? The token is not yet created, so present tense.